camel/storages/vectordb_storages/qdrant.py

```diff
@@ -0,0 +1,350 @@
+
+from camel.storages.vectordb_storages import (
+    BaseVectorStorage,
+    VectorDBQuery,
+    VectorDBQueryResult,
+    VectorDBStatus,
+    VectorRecord,
+)
+from camel.types import VectorDistance
+
+_qdrant_local_client_map: Dict[str, Tuple[Any, int]] = {}
+
+
+class QdrantStorage(BaseVectorStorage):
+    r"""An implementation of the `BaseVectorStorage` for interacting with
+    Qdrant, a vector search engine.
+
+    The detailed information about Qdrant is available at https://qdrant.tech/.
+
+    Args:
+        vector_dim (int): The dimenstion of storing vectors.
+        collection (Optional[str], optional): Name for the collection in the
+            Qdrant. If not provided, set it to the current time with iso
+            format. (default: :obj:`None`)
+        url_and_api_key (Optional[Tuple[str, str]], optional): Tuple containing
+            the URL and API key for connecting to a remote Qdrant instance.
+            (default: :obj:`None`)
+        path (Optional[str], optional): Path to a directory for initializing a
+            local Qdrant client. (default: :obj:`None`)
+        distance (VectorDistance, optional): The distance metric for vector
+            comparison (default: :obj:`VectorDistance.COSINE`)
+        delete_collection_on_del (bool, optional): Flag to determine if the
+            collection should be deleted upon object destruction.
+            (default: :obj:`False`)
+        **kwargs (Any): Additional keyword arguments for initializing
+            `QdrantClient`.
+
+    Notes:
+        - If `url_and_api_key` is provided, it takes priority and the client
+          will attempt to connect to the remote Qdrant instance using the URL
+          endpoint.
+        - If `url_and_api_key` is not provided and `path` is given, the client
+          will use the local path to initialize Qdrant.
+        - If neither `url_and_api_key` nor `path` is provided, the client will
+          be initialized with an in-memory storage (`":memory:"`).
+    """
+
+    def __init__(
+        self,
+        vector_dim: int,
+        collection: Optional[str] = None,
+        url_and_api_key: Optional[Tuple[str, str]] = None,
+        path: Optional[str] = None,
+        distance: VectorDistance = VectorDistance.COSINE,
+        delete_collection_on_del: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        try:
+            from qdrant_client import QdrantClient
+        except ImportError:
+            raise ImportError(
+                "Please install `qdrant-client` first. You can install it by "
+                "running `pip install qdrant-client`.")
+
+        self._client: QdrantClient
+        self._local_path: Optional[str] = None
+        self._create_client(url_and_api_key, path, **kwargs)
+
+        self.vector_dim = vector_dim
+        self.distance = distance
+        self.collection = collection or self._generate_collection_name()
+        self._assign_collection()
```

Comment:
`_assign_collection` is a bit confusing. How about calling it `_check_and_create_collection`?